lib/route.js

```diff
@@ -484,6 +484,21 @@ function buildRouting (options) {
       runPreParsing(null, request, reply)
     }
 
+    if (context.onClientAbort !== null) {
```

Comment:
Would it jump into race-condition? Since `onRequest`, `preParing` is already running.

Reply:
It is hard to say: this runs outside of lifecycle as the event can happen at any time.
I don't consider this hook part of the lifecycle as it is not bound to anything: the client disconnect can happen while server is receiving the request body or while it's sending the response.

Does it answer your question?

Reply:
> Does it answer your question?

Since the hooks provide `Request` and `Reply` object, it is easily for the user to misuse or misunderstanding how it behave.
I also afraid the `noop` as callback which just hide all the errors for `Promise`.

Reply:
Good point about Request/Reply, I should probably remove Reply since it makes no sense (as the socket is gone). Do you agree?

On the second one I get your point but I'm not familiar with internal hook error handling. What should I provide as alternative?

Reply:
> Do you agree?

Yes, until someone complain on it.

> What should I provide as alternative?

I would simply log the error since it is important for the dev to know something want wrong inside their handling.

Reply:
> I would simply log the error since it is important for the dev to know something want wrong inside their handling.

Agreed. I'll take care of this later today.

Reply:
@climba03003 I took care of this. How does it look now?